lib/modules/manager/helmv3/common.ts

```diff
@@ -1,19 +1,40 @@
 import { quote } from 'shlex';
 import upath from 'upath';
 
+import { logger } from '../../../logger';
+import { coerceArray } from '../../../util/array';
 import type { ExtraEnv } from '../../../util/exec/types';
 import { privateCacheDir } from '../../../util/fs';
+import { addSecretForSanitizing } from '../../../util/sanitize';
+import { fromBase64 } from '../../../util/string';
+import { ecrRegex, getECRAuthToken } from '../../datasource/docker/ecr';
 import type { RepositoryRule } from './types';
 
-export function generateLoginCmd(
+export async function generateLoginCmd(
   repositoryRule: RepositoryRule,
   loginCMD: string
-): string | null {
-  const { username, password } = repositoryRule.hostRule;
-  if (username && password) {
+): Promise<string | null> {
+  const { hostRule, repository } = repositoryRule;
+  const { username, password } = hostRule;
+  if (ecrRegex.test(repository)) {
+    logger.trace({ repository }, `Using ecr auth for Helm registry`);
+    const [, region] = coerceArray(ecrRegex.exec(repository));
+    const auth = await getECRAuthToken(region, hostRule);
+    if (auth) {
+      const [username, password] = fromBase64(auth).split(':');
+      if (username && password) {
+        addSecretForSanitizing(username);
+        addSecretForSanitizing(password);
+        return `${loginCMD} --username ${quote(username)} --password ${quote(
+          password
+        )} ${repository}`;
+      }
+    }
+  } else if (username && password) {
```

Comment:
```suggestion
  }
  if (username && password) {
```

Reply:
This is another change that makes the code differ from the `docker` datasource, which doesn't fall back to username + password auth if an ECR token can't be acquired. Again, I've made the change in this PR, just wanted to highlight the behaviour difference and make sure that's what you're after 👍